### PR TITLE
chore(storybook): re-enable for pull requests

### LINKED ISSUE #1100: Re-enable Storybook
### Please describe the task that needs to be done

We should look to re-enable Storybook to continue to ensure that our code base has the separation of concerns for visual components.

- [x] Re-add Storybook
- [x] Fix Storybook (again) after class implementation, unless #1233 gets resolved beforehand
- [x] Add MSW to mock network requests where necessary (see #1258, hopefully we can use this mocking for both types of tests)
- [x] Add stories for components
- [x] Add documentation for components, where possible
- [x] Update relevant API keys where necessary
- [x] Ensure Storybook stories are run on each PR
- ~~Test the stories, whether through React Testing Library or Cypress component testing~~ TBD, might add too much overhead

Tests should only be run against the `main` branch and primary repo, not for forks. This could be good to do once #1183 and other refactoring of stateful components is done.

### PR DESCRIPTION
**Pull request by kahboom**
This PR adds support for building stories on pull requests to `kaoto-ui` from any fork or branch, excluding Dependabot PRs.

Resolves #1100

Storybook no longer provides its own Babel configuration, so we have had to introduce it. This should not cause build issues as we are using Webpack and `ts-loader` for transpilation for the project.

Follow-up PR will include Storybook MSW add-on for mocking network requests.

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -1 +1,2 @@
+CHROMATIC_PROJECT_TOKEN=14f589a59855
```
Comment (lordrip) [inline .env.example:1]: Although is not ideal, we agreed on having this token shared among contributors to allow pull request checks from forks.

More info: https://www.chromatic.com/docs/github-actions#run-chromatic-on-external-forks-of-open-source-projects

File: src/components/PlusButtonEdge.tsx [hunk 2]
```diff
@@ -1,18 +1,23 @@
...
 import { PlusIcon } from '@patternfly/react-icons';
 import { ReactNode } from 'react';
-import { EdgeText, getBezierPath, Position, useReactFlow } from 'reactflow';
+import { EdgeText, getBezierPath, Position } from 'reactflow';
 
 const foreignObjectSize = 40;
 
 export interface IPlusButtonEdge {
-  data?: any;
+  data?: {
```
Comment (lordrip) [inline src/components/PlusButtonEdge.tsx:15]: [curious-question]: What should be the right type for this component? I couldn't find a type combining `showBranchesTab` and `sourceStepNode` 🤔

Comment (lordrip) [reply src/components/PlusButtonEdge.tsx:15]: Ah, I see it now: https://github.com/KaotoIO/kaoto-ui/pull/1805/files#diff-3c58cb917b8929dfa554c2eca4f1e3cd89c20bc0b02d42cce794aa3f9b88d24f

**Review by lordrip — DISMISSED** [review]

**Review by igarashitm — DISMISSED** [review]
:tada:

**Comment by kahboom** [comment]
E2E tests are being flaky again, particularly for Chrome and Edge :(

**Pull request merged**